scripts/build.ps1

```diff
@@ -409,15 +427,29 @@ function CMake-Build {
 #                     Main Execution                         #
 ##############################################################
 
-if ($Platform -eq "winkernel") {
-    # Restore Nuget packages.
-    Log "Restoring packages..."
-    msbuild cxplat.kernel.sln -t:restore /p:RestorePackagesConfig=true /p:Configuration=$Config /p:Platform=$Arch
+if ($BuildToolSet -eq "vs") {
+    if ($Platform -eq "winkernel") {
+        # Restore Nuget packages.
+        Log "Restoring packages..."
+        msbuild cxplat.kernel.sln -t:restore /p:RestorePackagesConfig=true /p:Configuration=$Config /p:Platform=$Arch
 
-    if (!$ConfigureOnly) {
-        # Build the code.
-        Log "Building..."
-        msbuild cxplat.kernel.sln /m /p:Configuration=$Config /p:Platform=$Arch
+        if (!$ConfigureOnly) {
+            # Build the code.
+            Log "Building..."
+            msbuild cxplat.kernel.sln /m /p:Configuration=$Config /p:Platform=$Arch
+        }
+    } elseif ($Platform -eq "windows") {
+        # Restore Nuget packages.
```

Comment:
Just set a $solutionFile variable based on $Platform value and we can re-use the same code block for both winkernel and windows vs build?